setup.sh.example

```diff
@@ -1,42 +1,130 @@
-#!/bin/sh
+#!/bin/bash
 #
-# setup.sh.example
+# This script can be used to safely inject secrets into the Docker
+# build process.
 #
-# This script is then used in the Docker image build process as a secret.
-# It should set up the default remote and authenticate if necessary.
+# Configure each `setup_` function defined as required and uncomment
+# them at the bottom. Then save the file as `setup.sh`, which `Makefile.docker`
+# will automatically add as a secret to the Docker image build process.
 #
-# Copy this file to setup.sh and modify it to allow Docker configure Conan in
-# the Docker build process. If this file is absent, the default Conan settings
-# are used.
-#
-# Authenticate with the default remote using the correct username and password.
-# This should run without any user interaction.
 
+# If any part of this script fails, the build process will too.
 set -e
 
-export CONAN_REMOTE="https://artifactory.example.com/artifactory/api/conan/cloe-conan-local"
-export CONAN_REMOTE_VERIFY_SSL="True"
-export CONAN_LOGIN_USERNAME=
-export CONAN_PASSWORD=
+setup_hosts() {
+    # If you need to update /etc/hosts, you can use `--add-host` on the command line
+    # or you can add them here.
 
-export VI_LIC_SERVER="vtd-licenses.example.com"
+    local HOSTS=(
+        "93.184.216.34 example.com"
+    )
 
-# Don't try to set up remotes if --network=none.
-if [ "$(ls /sys/class/net)" != "lo" ]; then
-    # Set the request timeout to 360 seconds to work-around slow servers.
-    conan config set general.request_timeout=360
+    cat /etc/hosts >> /tmp/hosts
+    for line in $HOSTS; do
+        echo $line >> /tmp/hosts
+    done
+    mount -o bind /tmp/hosts /etc/hosts
 
-    if [ "${CONAN_REMOTE}" != "" ]; then
-        conan remote clean
-        conan remote add default "${CONAN_REMOTE}" "${CONAN_REMOTE_VERIFY_SSL}"
-    fi
+    CLEANUP_FUNCTIONS+=(cleanup_hosts)
+}
+
+cleanup_hosts() {
+    umount /etc/hosts
+}
+
+setup_ssh() {
+    # If you need SSH for any reason, you can make your SSH agent available to Docker
+    # by passing the arguments `--ssh default=$SSH_AUTH_SOCK`.
+    # You can then use ssh-scankey to add known hosts or hosts you want to fetch
+    # things from.
+
+    local HOSTS=(
+        "-p 80 github.com"
+    )
+
+    for host in $HOSTS; do
+        ssh-keyscan $host >> ~/.ssh/known_hosts
```

Comment:
Minor observation: if this function is executed several times then we will have duplicated host ssh key information.

Reply:
Fixed.